evalml/automl/automl_algorithm/evalml_algorithm.py

```diff
@@ -0,0 +1,367 @@
+import inspect
+
+import numpy as np
+from skopt.space import Categorical, Integer, Real
+
+from .automl_algorithm import AutoMLAlgorithm
+
+from evalml.automl.utils import get_hyperparameter_ranges
+from evalml.model_family import ModelFamily
+from evalml.pipelines.components import (
+    RFClassifierSelectFromModel,
+    RFRegressorSelectFromModel,
+)
+from evalml.pipelines.components.transformers.column_selectors import (
+    SelectColumns,
+)
+from evalml.pipelines.components.utils import (
+    get_estimators,
+    handle_component_class,
+)
+from evalml.pipelines.utils import (
+    _make_stacked_ensemble_pipeline,
+    make_pipeline,
+)
+from evalml.problem_types import is_regression
+
+
+class EvalMLAlgorithm(AutoMLAlgorithm):
+    """An automl algorithm that consists of two modes: fast and long. Where fast is a subset of long.
+
+    1. Naive pipelines:
```

Comment:
Thought it would be nice to list it out here - can remove as well.

Reply:
Note: this current algorithm doesn't have any feature engineering steps as that is blocked by #2439.